tracer/src/Datadog.Trace/FeatureFlags/Exposure/ExposureApi.cs

```diff
@@ -0,0 +1,163 @@
+    {
+        internal static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ExposureApi));
+
+        public const string ExposurePath = "api/v2/exposure";
+        private readonly TaskCompletionSource<bool> _processExit = new();
+        private readonly TimeSpan _sendInterval = TimeSpan.FromSeconds(10);
+        private readonly Queue<ExposureEvent> _exposures = new Queue<ExposureEvent>();
+        private IApiRequestFactory? _apiRequestFactory = null;
+        private Dictionary<string, string> _context = new Dictionary<string, string>();
+        private int _started = 0;
+
+        private ExposureApi(TracerSettings tracerSettings)
+        {
+            UpdateSettings(tracerSettings.Manager.InitialExporterSettings, tracerSettings.Manager.InitialMutableSettings);
+            tracerSettings.Manager.SubscribeToChanges(settingChanges => UpdateSettings(settingChanges.UpdatedExporter ?? settingChanges.PreviousExporter, settingChanges.UpdatedMutable ?? settingChanges.PreviousMutable));
+        }
+
+        private void UpdateSettings(ExporterSettings exporterSettings, MutableSettings settings)
+        {
+            Log.Debug("ExposureApi::UpdateSettings -> Applyting settings");
+            var apiRequestFactory = AgentTransportStrategy.Get(
+                exporterSettings,
+                productName: "FeatureFlags exposure",
+                tcpTimeout: TimeSpan.FromSeconds(5),
+                AgentHttpHeaderNames.MinimalHeaders,
+                () => new MinimalAgentHeaderHelper(),
+                uri => uri);
+            Interlocked.Exchange(ref _apiRequestFactory, apiRequestFactory);
+
+            var context = new Dictionary<string, string>
+            {
+                { "service", settings.ServiceName ?? "unknown" },
+                { "env", settings.Environment ?? "unknown" },
+                { "version", settings.ServiceVersion ?? "unknown" }
+            };
+            Interlocked.Exchange(ref _context, context);
+        }
+
+        public static ExposureApi Create(TracerSettings tracerSettings)
+        {
+            return new ExposureApi(tracerSettings);
+        }
+
+        public void Dispose()
+        {
+            try { _processExit.TrySetResult(true); }
+            catch { }
+        }
+
+        public void TryToStartSendLoopIfNotStarted()
+        {
+            if (_started != 0 || Interlocked.Exchange(ref _started, 1) != 0)
+            {
+                return;
+            }
+
+            _ = Task.Run(SendLoopAsync).ContinueWith(t => { Log.Error(t.Exception, "FeatureFlags Exposure send loop failed"); }, TaskContinuationOptions.OnlyOnFaulted);
+        }
+
+        private async Task SendLoopAsync()
+        {
+            Log.Debug("ExposureApi::SendLoopAsync -> Enter");
+            while (!_processExit.Task.IsCompleted)
+            {
+                try
+                {
+                    var apiRequestFactory = _apiRequestFactory;
```

Comment:
Perfect, was going to check that we handled the update correctly, this looks good to me 👍